lib/iris/util.py

```diff
@@ -1828,3 +1828,30 @@ def equalise_attributes(cubes):
         for key in list(cube.attributes.keys()):
             if key not in common_keys:
                 del cube.attributes[key]
+
+
+def _remove_ancils_and_cms_mapping_dims(cube, dims):
+    """
+    Remove ancillary variables and cell measures that map to specific dimensions.
+
+    Returns a cube copy with (possibly) some cell-measures and ancillary variables removed.
+
+    To be used by operations that modify or remove dimensions.
+    Note: does nothing to (aux)-coordinates.  Those would be handled explicitly by the calling operation.
+
+    """
+    reduced_cube = cube.copy()
+
+    # Remove any ancillary variables that span the dimension(s) being collapsed
+    for ancil in reduced_cube.ancillary_variables():
```

Comment:
You could iterate over `(reduced_cube.ancillary_variables() + reduced_cube.cell_measures())` here, to reduce this to a single loop.  I think that should be ok, as the docstrings specify that these methods do return _lists_ (not iterables).

Reply:
I think doing so would require some kind of generic `cube_dims` and `remove_from_cube` methods on all dimensional metadata. Perhaps this is worthwhile anyway, but it seems non-trivial. Perhaps it would be worth a separate issue adding a bunch of methods to all dimensional metadata objects?

Reply:
Sorry, I had glossed over the internal differences in the loops.
We do have the cube_dims function now, but the generic "cube.remove" is I think a step too far.
So having seen that, I withdraw the suggestion :  I don't think we can justify the further complication !